lib/command/Command.js

```diff
@@ -279,17 +279,29 @@ class Command {
         return true;
     }
 
-    process(args, msg) {
+    async process(args, msg) {
         var shouldDelete = this.deleteCommand && msg.channel.guild && msg.channel.permissionsOf(msg._client.user.id).has("manageMessages");
 
         if(this.hooks.preCommand) {
-            this.hooks.preCommand(msg, args);
+            if (this.hooks.preCommand(msg, args) !== undefined) {
```

Comment:
Unless I'm mistaken, this unnecessarily runs this hook twice. It's preferable to run it, then do something different based on the output (or use `Promise.resolve()`).